dev-infrastructure/templates/mgmt-cluster.bicep

```diff
@@ -101,7 +101,7 @@ module maestroConsumer '../modules/maestro/maestro-consumer.bicep' = if (deployM
     )[0].uamiClientID
     namespace: maestroNamespace
     maestroInfraResourceGroup: maestroInfraResourceGroup
-    maestroConsumerName: mgmtCluster.outputs.aksClusterName
+    maestroConsumerName: resourceGroup().name
```

Comment:
i'm ok with this. since this names is passed to several others systems (eventgrid, maestro), we would need to make sure that the resourcegroup name consists only of certain characeters. the following seems to be safe

* max 90 chars long
* contains only characters, numbers, underscores and hyphens

is this going to compatible with what you have in mind RG naming wise?
in any case, we should have some validation in place

Reply:
yes, the naming convention for the RG is ```hcp-underlay-${region}-mgmt-${stamp number}```. so it should be within the boundary.

Reply:
do you want to add some validation for this in this PR? if not, we can also add it in an upcoming PR

Reply:
just added the validation